react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/OkHttpHeaderInterceptor.java

```diff
@@ -0,0 +1,35 @@
+package org.wordpress.mobile.WPAndroidGlue;
+
+import android.util.Log;
+
+import org.wordpress.mobile.WPAndroidGlue.WPAndroidGlueCode.OnAuthHeaderRequestedListener;
+
+import java.io.IOException;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class OkHttpHeaderInterceptor implements Interceptor {
+
+    private static final String AUTHORIZATION_HEADER_KEY = "Authorization";
+
+    private OnAuthHeaderRequestedListener mOnAuthHeaderRequestedListener;
+
+    void setOnAuthHeaderRequestedListener(OnAuthHeaderRequestedListener onAuthHeaderRequestedListener) {
+        mOnAuthHeaderRequestedListener = onAuthHeaderRequestedListener;
+    }
+
+    @Override
+    public Response intercept(Chain chain) throws IOException {
+        Request.Builder builder = chain.request().newBuilder();
+
+        String authHeader = mOnAuthHeaderRequestedListener.onAuthHeaderRequested(chain.request().url().toString());
+        if (authHeader.length() > 0) {
```

Comment:
Can we maybe use `if (!TextUtils.isEmpty(authHeader))` here? this would avoid a NPE in case `authHeader` was null

Reply:
Hey @mzorz, yup we can do that.